src/processes/helper/createEmptyChannel.js

```diff
@@ -0,0 +1,29 @@
+const {CHANNEL_TYPE_STRING} = require('../../utils');
+const {LogError} = require('../../databases/models');
+const StreamChat = require('stream-chat').StreamChat;
+
+const createEmptyChannel = async (userId, usersFollowed) => {
+  try {
+    const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+
+    await Promise.all(
+      usersFollowed.map(async (userFollowed) => {
+        const members = [userId, userFollowed];
+        const channel = await client.channel(CHANNEL_TYPE_STRING.CHAT, {
+          members,
+          created_by_id: userId
+        });
+
+        const createdChannel = await channel.create();
+        console.info('Created Channel:', createdChannel.channel.id);
+      })
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `createEmptyChannel` creates a new channel for each user followed by the given user. However, it doesn't handle the case where a channel between two users already exists. This could lead to duplicate channels between the same users. To avoid this, you should check if a channel already exists before creating a new one.

```diff
+    await Promise.all(
+      usersFollowed.map(async (userFollowed) => {
+        const members = [userId, userFollowed];
+        // Check if a channel already exists
+        const existingChannel = client.queryChannels({
+          members: { $in: members },
+          type: CHANNEL_TYPE_STRING.CHAT,
+        });
+
+        if (existingChannel.length === 0) {
+          const channel = await client.channel(CHANNEL_TYPE_STRING.CHAT, {
+            members,
+            created_by_id: userId
+          });
+
+          const createdChannel = await channel.create();
+          console.info('Created Channel:', createdChannel.channel.id);
+        }
+      })
+    );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->